cli/tests/end_to_end/extension.rs

```diff
@@ -1,128 +1,97 @@
-use std::fs;
-use std::path::Path;
-
-use assert_cmd::assert::Assert;
-use assert_cmd::Command;
-use tempfile::TempDir;
-
-use crate::end_to_end;
-
-const HEADER: &str = "import { PhylumApi } from 'phylum';";
+use crate::common::{create_lockfile, create_project, TestCli};
 
 /// Test Phylum API functions.
 #[tokio::test]
 pub async fn api() {
-    let tempdir = TempDir::new().unwrap();
-    let tempdir = tempdir.path();
-
-    let lockfile = end_to_end::create_lockfile(tempdir);
-    let config = end_to_end::create_config(tempdir);
-    let project = end_to_end::create_project().await;
-
-    with_extension(&config, "console.log(await PhylumApi.getUserInfo())", |assert| {
-        assert.success().stdout(predicates::str::contains("email"));
-    });
-
-    with_extension(&config, "console.log(await PhylumApi.getAccessToken())", |assert| {
-        assert.success().stdout(predicates::str::contains("ey"));
-    });
-
-    with_extension(&config, "console.log(await PhylumApi.getRefreshToken())", |assert| {
-        assert.success().stdout(predicates::str::contains("ey"));
-    });
-
-    with_extension(
-        &config,
-        "console.log(await PhylumApi.getPackageDetails('express', '4.18.1', 'npm'))",
-        |assert| {
-            assert.success().stdout(predicates::str::contains("vulnerability: 1"));
-        },
-    );
-
-    let project_details = format!("console.log(await PhylumApi.getProjectDetails({project:?}))");
-    with_extension(&config, &project_details, |assert| {
-        assert.success().stdout(predicates::str::contains("name: \"integration-tests\""));
-    });
-
-    let parse_lockfile = format!(
-        "const packages = await PhylumApi.parseLockfile({lockfile:?}, 'yarn');
-        console.log(packages);",
-    );
-    with_extension(&config, &parse_lockfile, |assert| {
-        assert.success().stdout("[ { name: \"accepts\", version: \"1.3.8\", type: \"npm\" } ]\n");
-    });
-
-    let analyze = format!(
-        "const jobId = await PhylumApi.analyze({lockfile:?}, {project:?});
-        console.log(await PhylumApi.getJobStatus(jobId));
-    "
-    );
-    with_extension(&config, &analyze, |assert| {
-        assert.success().stdout(predicates::str::contains("name: \"accepts\""));
-    });
+    let test_cli = TestCli::builder().with_config(true).build();
+
+    let lockfile = create_lockfile(test_cli.temp_path());
+    let project = create_project().await;
+
+    {
+        test_cli
+            .create_extension("console.log(await PhylumApi.getUserInfo())")
+            .run()
+            .success()
+            .stdout(predicates::str::contains("email"));
+    }
```

Comment:
What's the point of the scopes here?

We already have each individual API separated by whitespace, if separation is so important we need to have useless scopes, they should probably be moved into separate functions.

Unless the scopes here are necessary to make things compile?

Reply:
The `create_extension` method returns a RAII guard that uninstalls and deletes the extension once it goes out of scope. We could give a randomized name to the extension instead of the hardcoded `sample` like we had in the previous `end_to_end` tests but I felt the RAII guard and scoping were a viable solution as well, plus we don't have to worry about collisions and other similar concerns. We may want to document this, though.